### ISSUE TITLE
[Feature request] Add recommended videos based on what user watches, subs to, and searches

### ISSUE THREAD

**Issue by SudoLinux** [TeamNewPipe/NewPipe#893]
I know this feature could be a little bit privacy concerning to some, but, it would be nice to add a setting under "Settings > Content > Content of main page" called "recommended videos" that basically sum up your video history and even possibly search history and show recommended videos, like if you love watching videos about Linux/FOSS, it will show videos about that specific subject. Also for people that like to see latest videos with subscribers but also want that feature, try adding a new section to kiosk mode too, you can add one that shows "recommended videos/feed page" so not only does it show your subscribers latest videos but shows related videos (and maybe even show recommended videos from your subscribers too!). Anyway, this is a feature that would make me start using NewPipe even more, thank you so much for developing NewPipe and have a nice day :smile:

EDIT: I wondered if this would work WITHOUT the need of a privacy grabbing account, its not necessary to me and I can live without this, but it would be pretty cool.

**Comment by theScrabi** [comment]
Recommended is not possible with NewPipe. Because for this to work Youtube has to Identify you again, but NewPipe does not send enough information for that.

**Comment by SudoLinux** [comment]
@theScrabi How so, nothing to do with accounts if that's what you mean

**Comment by theScrabi** [comment]
I know but you'd have to save cookies for that, and NewPipe is not doing that.

The question is would we want to do that in order to allow that?

**Comment by SudoLinux** [comment]
@theScrabi that would be cool!

**Comment by wb9688** [comment]
@theScrabi: I wouldn't want that, but wasn't one of NewPipe's goals privacy?

**Comment by Cormak** [comment]
I wouldn't want this. I like just looking at the trending videos. The point of this app is to be not participate in Google's tracking while getting all those delicious videos.

**Comment by theScrabi** [comment]
Yes I'm not a big fan of it either.

**Comment by r2308145** [comment]
I dont like this idea either.. there are already "recommended" videos under next videos at the bottom of the video page

**Comment by TomasGutierrez0** [comment]
This could be possible while keeping privacy in mind but I don't know how much it could impact performance nor how accurate it would be. 
A "Recommend me videos" button which will only suggest videos when pressed.

Based on the last 10-20 searches and watch history, parse the title of the videos and searches, and take the first or two words most frequently used, then based on those word/s do a search but filter videos that have only been uploaded this month.

For example a normal search looks like this. 
https://www.youtube.com/results?search_query=Linux
When you show only uploaded in the last month.
https://www.youtube.com/results?sp=EgIIBA%253D%253D&search_query=Linux

I can definitely see problems with this such as most used words will be "the" "a" "is" stuff like that which could be added to some sort of dictionary so they can be deleted from the most used words list. 
I don't know how NewPipe do search queries so I don't know if it's feasible either.

I'd love to see what @theScrabi and others have to say about this if they are really interested in a recommended videos. **Personally I'm not interested in this** but if there is people interested they can take my idea.

**Comment by SudoLinux** [comment]
@TomasGutierrez0 Yesh, well I should have been more specific. I indeed care about privacy by a hunch, YouTube is the ONLY google service I limit myself to, startpage and others i use. I was thinking that this could be possible WITHOUT the need of an account, just take your search's and possibly just search them and combine them would be right. No google tracker needed. Sorry I needed to be more specific, I'll update the post.
